src/test/util_sds_check/util_sds_check.c

```diff
@@ -0,0 +1,185 @@
+	ASSERTeq(RCOUNTER(shutdown_state_reinit), 1);
+
+	return ret;
+}
+
+/* Pool_sds having an invalid checksum. */
+static int
+test_invalid_checksum()
+{
+	SET_SDS(Pool_sds, SDS_CUSTOM(USC, UUID, DIRTY_CLEAR, INVALID_VALUE));
+	SET_SDS(Curr_sds, SDS_CUSTOM(USC, UUID, DIRTY_CLEAR, CHECKSUM));
+
+	int ret = shutdown_state_check(&Curr_sds, &Pool_sds, &Rep);
+	ASSERTeq(ret, 0);
+	ASSERTeq(RCOUNTER(shutdown_state_reinit), 1);
+
+	return ret;
+}
+
+/*
+ * Test with curr_sds and pool_sds having the same values
+ * except for the dirty field.
+ */
+static int
+test_dirty_set()
+{
+	SET_SDS(Pool_sds, SDS_CUSTOM(USC, UUID, DIRTY_SET, CHECKSUM));
+	shutdown_state_checksum(&Pool_sds, &Rep);
+	SET_SDS(Curr_sds, SDS_CUSTOM(USC, UUID, DIRTY_CLEAR, CHECKSUM));
+
+	int ret = shutdown_state_check(&Curr_sds, &Pool_sds, &Rep);
+	ASSERTeq(ret, 0);
+	ASSERTeq(RCOUNTER(shutdown_state_reinit), 1);
+
+	return ret;
+}
+
+/* Pool_sds and Curr_sds having different uuid. */
+static int
+test_invalid_uuid()
+{
+	SET_SDS(Pool_sds, SDS_CUSTOM(USC, UUID ^ INVALID_VALUE, DIRTY_CLEAR,
+		CHECKSUM));
+	shutdown_state_checksum(&Pool_sds, &Rep);
+	SET_SDS(Curr_sds, SDS_CUSTOM(USC, UUID, DIRTY_CLEAR, CHECKSUM));
+
+	int ret = shutdown_state_check(&Curr_sds, &Pool_sds, &Rep);
+	ASSERTeq(ret, 0);
+	ASSERTeq(RCOUNTER(shutdown_state_reinit), 1);
+
+	return ret;
+}
+
+/*
+ * Pool_sds and Curr_sds having different uuid and
+ * Pool_sds.dirty is set.
+ */
+static int
+test_invalid_uuid_set_dirty()
+{
+	SET_SDS(Pool_sds, SDS_CUSTOM(USC, UUID ^ INVALID_VALUE, DIRTY_SET,
+		CHECKSUM));
+	shutdown_state_checksum(&Pool_sds, &Rep);
+	SET_SDS(Curr_sds, SDS_CUSTOM(USC, UUID, DIRTY_CLEAR, CHECKSUM));
+
+	int ret = shutdown_state_check(&Curr_sds, &Pool_sds, &Rep);
+	ASSERTeq(ret, 0);
+	ASSERTeq(RCOUNTER(shutdown_state_reinit), 1);
+
+	return ret;
+}
+
+/* Pool_sds and Curr_sds having different USC and the same UUID. */
+static int
+test_invalid_usc()
+{
+	SET_SDS(Pool_sds, SDS_CUSTOM(USC ^ INVALID_VALUE, UUID, DIRTY_CLEAR,
+		CHECKSUM));
+	shutdown_state_checksum(&Pool_sds, &Rep);
+	SET_SDS(Curr_sds, SDS_CUSTOM(USC, UUID, DIRTY_CLEAR, CHECKSUM));
+
+	int ret = shutdown_state_check(&Curr_sds, &Pool_sds, &Rep);
+	ASSERTeq(ret, 0);
+	ASSERTeq(RCOUNTER(shutdown_state_reinit), 1);
+
+	return ret;
+}
+
+/*
+ * Pool_sds and Curr_sds having different USC, the same UUID and
+ * Pool_sds.dirty is set.
+ */
+static int
+test_invalid_usc_set_dirty()
+{
+	SET_SDS(Pool_sds, SDS_CUSTOM(USC ^ INVALID_VALUE, UUID, DIRTY_SET,
+		CHECKSUM));
+	shutdown_state_checksum(&Pool_sds, &Rep);
+	SET_SDS(Curr_sds, SDS_CUSTOM(USC, UUID, DIRTY_CLEAR, CHECKSUM));
+
+	int ret = shutdown_state_check(&Curr_sds, &Pool_sds, &Rep);
+	ASSERTeq(ret, 1);
+	ASSERTeq(RCOUNTER(shutdown_state_reinit), 0);
+
+	return ret;
+}
+
+/* Pool_sds and Curr_sds having identical values (dirty==0). */
+static int
+test_happy_day()
+{
+	SET_SDS(Pool_sds, SDS_CUSTOM(USC, UUID, DIRTY_CLEAR, CHECKSUM));
+	shutdown_state_init(&Pool_sds, &Rep);
+
```

Comment:
`Curr_sds` is uninitialized.

```suggestion
SET_SDS(Curr_sds, SDS_CUSTOM(USC, UUID, DIRTY_CLEAR));
```

Reply:
Done.